### PR TITLE
⚡ Improve test summary formatting

### PR DESCRIPTION
**Pull request by PraneshASP**
## Motivation
Currently, the test result summary doesn't differentiate between the failed, passed and skipped tests visually. So it would be nice to have colored test summary similar to Hardhat. 

Also redundant summary is printed if any of the tests fail. It is no longer required as we already print aggregated test summary by default (Issue #5195).

## Solution
Used red, green and blue colors in test summary for the no.of failed, passed and skipped tests respectively. 
Also removed the redundant test summary output.  

### Before: 
Screenshot 2023-08-27 at 12 24 08 AM

### After: 
Screenshot 2023-08-27 at 12 25 48 AM

### CODE REVIEW

File: crates/forge/bin/cmd/test/mod.rs [hunk 1]
```diff
@@ -397,13 +397,6 @@ impl TestOutcome {
             }
             println!();
         }
-
-        let successes = self.successes().count();
-        println!(
-            "Encountered a total of {} failing tests, {} tests succeeded",
-            Paint::red(failures.to_string()),
-            Paint::green(successes.to_string())
-        );
```
Comment (Evalir) [inline crates/forge/bin/cmd/test/mod.rs:406]: hmm why remove this? don't think we should

File: crates/forge/bin/cmd/test/mod.rs [hunk 2]
```diff
@@ -472,12 +465,16 @@ fn format_aggregated_summary(
...
+        num_test_suites,
+        Paint::green(total_passed),
+        Paint::red(total_failed),
+        Paint::blue(total_skipped),
+        total_tests
     )
 }
 
-/// Lists all matching tests
+/// Lists all matching tests
```
Comment (Evalir) [inline crates/forge/bin/cmd/test/mod.rs:477]: extra whitespace

**Review by Evalir — CHANGES_REQUESTED** [review]
i like this! some comments

File: crates/forge/bin/cmd/test/mod.rs (hunk 1 above)
Comment (PraneshASP) [reply crates/forge/bin/cmd/test/mod.rs:406]: We already display the aggregated test summary for every runs, which was added in https://github.com/foundry-rs/foundry/pull/5266.

```bash
Ran 4 test suites: 52 tests passed, 2 failed, 0 skipped (54 total tests)
Failing tests:
....

Encountered a total of 2 failing tests, 52 tests succeeded
```

The output is redundant in-case of test failures, as the total failed and passed test counts are displayed once again at the end.  See attached screenshots above for reference.

File: crates/forge/bin/cmd/test/mod.rs (hunk 2 above)
Comment (PraneshASP) [reply crates/forge/bin/cmd/test/mod.rs]: removed: https://github.com/foundry-rs/foundry/pull/5731/commits/9cfeaa20d0240859ba3c8ebd516efdc4bf3fa6f1

File: crates/forge/bin/cmd/test/mod.rs (hunk 1 above)
Comment (Evalir) [reply crates/forge/bin/cmd/test/mod.rs:406]: Right, I understand your point—but I lean on keeping this for the sake of not making a breaking change on the tool's output. Let's keep it for now as it's something ppl expect to be there.

Comment (Evalir) [inline crates/forge/bin/cmd/test/mod.rs:406]: let's re-add this

**Review by Evalir — CHANGES_REQUESTED** [review]

File: crates/forge/bin/cmd/test/mod.rs (hunk 1 above)
Comment (PraneshASP) [reply crates/forge/bin/cmd/test/mod.rs:406]: Sounds good

**Comment by PraneshASP** [comment]
@Evalir Should I change the skipped tests count color to yellow, as the tool already uses it to denote `[SKIP]` for consistency? https://github.com/foundry-rs/foundry/blob/bff4ed912bb023d7bf9b20eda581aa4867a1cf70/crates/forge/bin/cmd/test/mod.rs#L434

**Comment by Evalir** [comment]
yeh i was debating if we should keep yellow or switch to blue here. Let's change to yellow for now @PraneshASP

File: crates/forge/bin/cmd/test/mod.rs (hunk 1 above)
Comment (PraneshASP) [reply crates/forge/bin/cmd/test/mod.rs]: Added here: https://github.com/foundry-rs/foundry/pull/5731/commits/e409e275015c65dd669cc835174c1ef4a7199389

**Review by Evalir — APPROVED** [review]

**Pull request merged**